### ISSUE TITLE
Lines duplicated in favicon html file each time task runs

### ISSUE THREAD

**Issue by FStop** [RealFaviconGenerator/grunt-real-favicon#27]
We're using the grunt task with a WordPress theme. We can't use the realFavicon plugin because of hosting restrictions and such.

For whatever reason the plugin would duplicate certain lines in the file we specified. Here's our grunt config:
```
realFavicon: {
	favicons: {
		src: 'assets/images/svg/our-logo.svg',
		dest: 'assets/images/favicons',
		options: {
			iconsPath: '<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/favicons/',
			html: [ 'parts/header/favicons.php' ],
			design: {
				ios: {
					pictureAspect: 'backgroundAndMargin',
					backgroundColor: '#ffffff',
					margin: '14%',
					appName: 'Our App Name'
				},
				desktopBrowser: {},
				windows: {
					pictureAspect: 'whiteSilhouette',
					backgroundColor: '#da532c',
					appName: 'Our App Name'
				},
				androidChrome: {
					pictureAspect: 'noChange',
					themeColor: '#ffffff',
					manifest: {
						name: 'Our App Name'',
						display: 'browser',
						orientation: 'notSet',
					}
				},
				safariPinnedTab: {
					pictureAspect: 'silhouette',
					themeColor: '#000000'
				}
			},
			settings: {
				scalingAlgorithm: 'Mitchell',
				errorOnImageTooSmall: false
			},
			versioning: {
				paramName: 'v',
				paramValue: new Date().getTime()
			}
		}
	}
},
```

The following lines get keep getting added to the top of the file each time the task runs:
```
<link rel="mask-icon" href="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/favicons/safari-pinned-tab.svg&v=1456847168781" color="#000000">
<meta name="apple-mobile-web-app-title" content="Our App Name">
<meta name="application-name" content="Our App Name">
```

Our stopgap solution was a [grunt-shell](https://github.com/sindresorhus/grunt-shell) task that empties the favicons.php file before the task runs:

```
shell: {
	favicons_pre: {
		command: 'echo "" > parts/header/favicons.php'
	}
},
```

**Comment by phbernard** [comment]
Sorry for answering so late.

The plugin is not neutral regarding the existing content of `parts/header/favicons.php` (or any other files listing in `html`). It processes it as HTML and tries to removes existing favicon markups in order to avoid duplicated entries. This is useful when you don't have an external file like `parts/header/favicons.php` to carry these markups, but only raw HTML pages with some existing icons.

Regarding your specific case, I have the feeling that this mechanism worked (previous markups were stripped) except for these three lines. Hum... I need to check this.

Anyway, your workaround is actually the right way to go: you don't way RFG plugin to do clever stuff about the previous favicon markups. You just want it to dump the new markups, plain and easy.

**Comment by pdugas** [comment]
I know this is an old one but I was recently bitten by the same issue.  The problem is with the default set of "remove" filters in rtg-api's injectFaviconMarkups function. I reported an [issue](https://github.com/RealFaviconGenerator/rfg-api/issues/11) there.  You could fix it here instead by passing in the extra entries.

        grunt.log.writeln('Injecting markup into file: ' + file);
        rfg.injectFaviconMarkups(file, favicon.favicon.html_code, {remove: [
                'link[rel="mask-icon"]',
                'meta[name="apple-mobile-web-app-title"]',
                'meta[name="application-name"]',
            ]}, function(error, code) {
          // Handle dynamic iconPathsCallback.

**Comment by pdugas** [comment]
Or maybe you could find a way to make the filters configurable...

**Comment by phbernard** [comment]
@pdugas Regarding your case, the issue should be fixed by v0.2.4, published a minute ago. Please visit https://github.com/RealFaviconGenerator/rfg-api/pull/17#issuecomment-581801771 for more information.
